src/Workspaces/Core/Portable/Diagnostics/DiagnosticProvider.cs

```diff
@@ -44,6 +45,11 @@ public enum Options
             /// Include semantic errors
             /// </summary>
             Semantic = 0x02,
+
+            /// <summary>
+            /// Include script semantic errors
+            /// </summary>
+            ScriptSemantic = 0x04,
```

Comment:
> ScriptSemantic = 0x04, [](start = 11, length = 23)

Is there a reason why have a special value for scripts? Why not just check Options.Semantic regardless of whether the file is a script or not.

Reply:
we don't want to show semantic errors for regular C# files.

Reply:
Could that be just a PerLanguageOption?  instead of needing an actual enum value here?

Reply:
Ping on this question.  Is there a need for this special enum value?

Reply:
Ping on this question. Is there a need for this special enum value?